File: product-service/src/main/java/com/intexsoft/webshop/productservice/service/VendorService.java
package com.intexsoft.webshop.productservice.service;

import com.intexsoft.webshop.productservice.dto.vendor.VendorCreateDto;
import com.intexsoft.webshop.productservice.dto.vendor.VendorUpdateDto;
import com.intexsoft.webshop.productservice.dto.vendor.VendorDto;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface VendorService {
    VendorDto createVendor(VendorCreateDto vendorCreateDto);

    VendorDto findVendorById(Long vendorId);

    List<VendorDto> findVendors(Pageable pageable);

    VendorDto updateVendor(Long vendorId, VendorUpdateDto vendorUpdateDto);

    void deleteVendorById(Long vendorId);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/service/VendorService.kt
package com.intexsoft.webshop.productservicekt.service

import com.intexsoft.webshop.productservicekt.dto.vendor.VendorCreateDto
import com.intexsoft.webshop.productservicekt.dto.vendor.VendorDto
import com.intexsoft.webshop.productservicekt.dto.vendor.VendorUpdateDto
import org.springframework.data.domain.Pageable

interface VendorService {
    fun createVendor(vendorCreateDto: VendorCreateDto): VendorDto
    fun findVendorById(vendorId: Long): VendorDto
    fun findVendors(pageable: Pageable): List<VendorDto>
    fun updateVendor(vendorId: Long, vendorUpdateDto: VendorUpdateDto): VendorDto
    fun deleteVendorById(vendorId: Long)
}